src/codeLens.ts

```diff
@@ -84,6 +89,44 @@ export class LensProvider implements vscode.CodeLensProvider
     }
 }
 
+export async function code_lens_execute(code_lens: string, range: any) {
+    if(custom_code_lens) {
+        const auto_submit = custom_code_lens[code_lens]["auto_submit"];
+        let messages = custom_code_lens[code_lens]["messages"];
+        const start_of_line = new vscode.Position(range.start.line, 0);
+        const end_of_line = new vscode.Position(range.end.line + 1, 0);
+        const block_range = new vscode.Range(start_of_line, end_of_line);
+        let text = vscode.window.activeTextEditor!.document.getText(block_range);
+        if(!auto_submit) {
+            if (global && global.side_panel && global.side_panel._view) {
+                const message = setInputValue(text);
+                global.side_panel._view.webview.postMessage(message);
+            }
+            vscode.commands.executeCommand('refactaicmd.callChat', '');
```

Comment:
`vscode.commands.executeCommand('refactaicmd.callChat', '');` shouldn't be needed any more, if chat receives `setInputValue` or `addInputValue` it'll open the chat page